### DISCUSSION TITLE
How would you use this template in a monorepo (of Python projects)?

### DISCUSSION THREAD

**Discussion by cdinu** [UCL-ARC/python-tooling#53] [Q&A] [answered]
I have good experience with JavaScript monorepos.

Has anyone any experience with multiple autonomous projects in the same repo?

**Comment by paddyroddy** [comment, marked as answer]
So to answer your question directly, you could do something like this https://discuss.python.org/t/multiple-related-programs-one-pyproject-toml-or-multiple-projects/17427/2. So you could use the `cookieninja` template to make one and then move `src` into `project1/src`, then `cp -r project1/src project2/src` etc.

However, particularly if you're going to be pushing to PyPI etc. then I would probably recommend one package per repo. Just to keep things simple.

Another question is whether they are actually separate packages or modules. One package can have multiple modules, which are technically separate. For example, you may have seen
```python
from matplotlib import pyplot as plt
```
So here we are using the `pyplot` module. But `matplotlib` has other modules as can be seen here https://matplotlib.org/stable/py-modindex.html

**Reply by cdinu** [reply]
Thank you! This looks like a good way to go.

It is for an internal project (tentatively). That project used to have 43 (!) repositories, when we were handed it over; now we have three python repositories that we work with with some light interdependencies in between them.

I was exploring the idea having a single repository for all three, where we could have the privilege of safely sharing some code (e.g. getting a configuration setting or some data classes, fixtures/mocks, or connecting to the shared database). I haven't talked to the rest of the team yet, before I knew there were no big red flags.
